herddb-core/src/main/java/herddb/sql/CalcitePlanner.java

```diff
@@ -209,13 +212,18 @@ public void clearCache() {
     public ExecutionPlan plan(String defaultTableSpace, Statement stmt, boolean scan, boolean returnValues, int maxRows) {
         return fallback.plan(defaultTableSpace, stmt, scan, returnValues, maxRows);
     }
-
+    
+    static final boolean isDDL(String query) {
+        return USE_DDL_PARSER.matcher(query).matches();
+    }
+        
     @Override
-    public TranslatedQuery translate(String defaultTableSpace, String query, List<Object> parameters, boolean scan, boolean allowCache, boolean returnValues, int maxRows) throws StatementExecutionException {
-        query = SQLPlanner.rewriteExecuteSyntax(query);
-        if (query.matches("(?i)(EXECUTE|CREATE|DROP|ALTER|TRUNCATE).*")) {        
+    public TranslatedQuery translate(String defaultTableSpace, String query, List<Object> parameters, boolean scan, boolean allowCache, boolean returnValues, int maxRows) throws StatementExecutionException {        
+        if (isDDL(query)) {        
+            query = SQLPlanner.rewriteExecuteSyntax(query);
             return fallback.translate(defaultTableSpace, query, parameters, scan, allowCache, returnValues, maxRows);
         }
+        System.out.println("NOT DDL:'"+query+"'");
```

Comment:
This should be removed or added to a logger

Reply:
removed